### PR TITLE
New package: ezstream-1.0.2

### LINKED ISSUE #39345: Package request:
### Package name

ezstream

### Package homepage

https://icecast.org/ezstream/

### Description

Ezstream is a command line source client for media streams, primarily for streaming to Icecast servers.

It allows the creation of media streams based on input from files or standard input that is piped through an optional external de- and encoder. As every part of this chain is highly configurable, ezstream can be useful in a large number of streaming setups.

It uses libshout to communicate with streaming servers and currently supports Ogg, MP3, WebM, and Matroska streams using the HTTP, ICY, and RoarAudio protocols. It uses TagLib to read and manage metadata in numerous media files.

Ezstream is free software and licensed under the GNU General Public License.

### Dependencies

Ezstream depends on:

 - [libshout](https://www.icecast.org/download/) version 2.2.x or newer
 - [libxml](http://xmlsoft.org/) version 2.x
 - [TagLib](https://taglib.github.io/) for C version 1.x (1.4 or newer
   recommended)

Ezstream optionally uses:

 - [libiconv](https://www.gnu.org/software/libiconv/) on systems where
   `iconv()` is not available in libc, for basic non-ASCII charset support in
   metadata and filenames

Building ezstream depends on:

 - [check](https://libcheck.github.io/check) unit testing framework for C

### Does the requested package meet the package requirements?

System, Compiled

### Is the requested package released?

Yes

### PR DESCRIPTION
**Pull request by misuchiru03**
Closes #39345

#### Testing the changes
- I tested the changes in this PR: **briefly**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-GLIBC)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl
  - armv7l
  - armv6l-musl

### CODE REVIEW

File: srcpkgs/ezstream/template [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+checksum=11de897f455a95ba58546bdcd40a95d3bda69866ec5f7879a83b024126c54c2a
+
+#if [ "${XBPS_LIBC}" = musl ]; then
+#	hostmakedepends+=" musl-legacy-compat"
+#fi
+
+if [ "${XBPS_TARGET_LIBC}" = musl ]; then
+	makedepends+=" musl-legacy-compat"
+else
+	makedepends+=" glibc-devel"
```
Comment (classabbyamp) [inline srcpkgs/ezstream/template:23]: why is this needed?

File: srcpkgs/ezstream/template [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Template file for 'ezstream'
+pkgname=ezstream
+version=1.0.2
+revision=1
+build_style=gnu-configure
+hostmakedepends="gettext-devel automake pkg-config libtool"
+makedepends="check-devel libshout-devel libxml2-devel taglib-devel"
+depends="libshout libxml2 taglib"
```
Comment (classabbyamp) [inline srcpkgs/ezstream/template:8]: these should be auto-detected

File: srcpkgs/ezstream/template (hunk 1 above)
Comment (misuchiru03) [reply srcpkgs/ezstream/template:23]: glibc-devel was there for `sys/queue.h`

**Comment by misuchiru03** [comment]
Once I added the TLS support in libshout, this package fails at libshout checks.
Without openssl in libshout, ezstream will fail the TLS check.

Confirmed with removing the updated libshout package in `binpkgs` and libshout is detected.

Should we have two separate PRs for that?

File: srcpkgs/ezstream/template (hunk 2 above)
Comment (misuchiru03) [reply srcpkgs/ezstream/template:8]: It is indeed detected. I've removed it.

File: srcpkgs/ezstream/template (hunk 1 above)
Comment (classabbyamp) [reply srcpkgs/ezstream/template:23]: glibc-devel should be installed already for glibc-targeting builds

Comment (misuchiru03) [reply srcpkgs/ezstream/template:23]: Gotcha. Removed.
